File: Ejercicios Conexion BD/Go/MongoDB.go
// Instalar paquete MongoDB para Go
// go get go.mongodb.org/mongo-driver/mongo
// go get go.mongodb.org/mongo-driver/bson

package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

func main() {
	// Configurar contexto con timeout
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	// Conectar a MongoDB
	client, err := mongo.Connect(ctx, options.Client().ApplyURI("mongodb://localhost:27017"))
	if err != nil {
		log.Fatal("❌ Error de conexión:", err)
	}
	defer client.Disconnect(ctx)

	fmt.Println("✅ Conexión exitosa")
	fmt.Printf("🔎 Tipo de objeto de conexión: %T\n", client)

	// Listar bases de datos
	databases, err := client.ListDatabaseNames(ctx, bson.M{})
	if err != nil {
		log.Fatal("❌ Error al listar bases de datos:", err)
	}
	fmt.Println("📁 Bases de datos disponibles:", databases)

	// Acceder a la base de datos y colección
	db := client.Database("ecommerce3066478")
	collection := db.Collection("Clientes1")

	// Documento a insertar
	documento := bson.D{
		{Key: "Nombre", Value: "Fernanda"},
		{Key: "Correo", Value: "fernanda.lopez@example.com"},
		{Key: "Telefono", Value: "3129876543"},
		{Key: "Direccion", Value: "Calle 45 #67-89"},
		{Key: "Creado_desde", Value: "Go"},
	}
	

	// Insertar documento
	resultado, err := collection.InsertOne(ctx, documento)
	if err != nil {
		log.Fatal("❌ Error al insertar documento:", err)
	}
	fmt.Printf("📌 Documento insertado con ID: %v\n", resultado.InsertedID)

	fmt.Println("❗ Conexión cerrada")
}

File: Ejercicios Conexion BD/Javascript/MongoDB.js
// Instalar paquetes necesarios
// Ejecutar en la terminal: npm install mongodb

// Importamos la librería mongodb
const { MongoClient } = require('mongodb');

// URL de conexión a MongoDB
const url = "mongodb://localhost:27017/";

// Nombre de la base de datos
const dbName = "ecommerce3066478";

// Función principal para ejecutar la conexión y operaciones
async function main() {
    // Creamos una instancia del cliente MongoDB
    const client = new MongoClient(url);

    try {
        // Conectamos al servidor
        await client.connect();
        console.log("✅ Conexión exitosa");

        // Imprimimos el tipo de objeto creado
        console.log(`🔍 Tipo de objeto de conexión: ${client.constructor.name}`);

        // Listar todas las bases de datos disponibles en el servidor
        const databasesList = await client.db().admin().listDatabases();
        console.log("📁 Bases de datos disponibles: ", databasesList.databases.map(db => db.name));

        // Creamos o accedemos a la base de datos llamada "ecommerce3066478"
        const db = client.db(dbName);

        // Creamos o accedemos a la colección llamada "Clientes1" dentro de la base de datos
        const collection = db.collection("Clientes1");

        // Documento que se insertará en la colección
        const documento = {
            Nombre: "Bob", 
            Correo: "bob456@gmail.com",
            telefono: "3216549870",
            Direccion: "av 15 #45 - 67",
            "Creado desde": "JavaScript"
        };

        // Insertamos el documento en la colección
        const resultado = await collection.insertOne(documento);

        // Imprimimos el ID del documento insertado
        console.log(`📌 Documento insertado con ID: ${resultado.insertedId}`);

    } catch (err) {
        console.error("❌ Error:", err);
    } finally {
        // Cerramos la conexión con el servidor
        await client.close();
        console.log("❗ Conexión cerrada");
    }
}

// Ejecutamos la función principal
main().catch(console.error);
